internal/worker/worker.go

```diff
@@ -0,0 +1,54 @@
+package worker
+
+import (
+	"context"
+	"sync"
+)
+
+type Task interface {
+	Process()
+}
+
+type WorkerPool struct {
+	taskChan    chan Task
+	wg          sync.WaitGroup
+	concurrency int
+}
+
+func NewWorkerPool(concurrency int) *WorkerPool {
+	return &WorkerPool{
+		concurrency: concurrency,
+		taskChan:    make(chan Task),
```

Comment:
It would be better to choose a buffered channel here with the capacity equal to the number of workers

Reply:
You haven't fixed it here, have you ?

Reply:
For some reason, I missed this bit. It was done now.